lib/instrumentation/modules/knex.js

```diff
@@ -77,5 +72,19 @@ module.exports = function (Knex, agent, { version, enabled }) {
     }
   }
 
-  return Knex
+  return function wrappedKnex () {
+    const knexInstance = Knex.apply(null, arguments)
+
+    if (knexInstance && knexInstance.client) {
+      const QUERY_FNS = ['queryBuilder', 'raw']
+      agent.logger.debug('shimming knexInstance.client.runner')
+      shimmer.wrap(knexInstance.client, 'runner', wrapRunner)
+      agent.logger.debug('shimming Knex.Client.prototype functions: %j', QUERY_FNS)
+      shimmer.massWrap(knexInstance.client, QUERY_FNS, wrapQueryStartPoint)
+    } else {
+      agent.logger.debug('could not shim Knex')
+    }
+
+    return knexInstance
+  }
```

Comment:
As we discussed in a call, we'll sit on this for a little while until I have a chance to try it out with the ESM work. It is possible this new way of instrumenting will break with ESM instrumention. It is possible it will work.

Reply:
Yup, it works fine with the ESM wrapping. I think you can go ahead and look at the "nit" comments here if you like, then we can merge this.

Reply:
thanks for having a look at it :)